alphagov/tsuru-performance-tests
# [98091996] Deploy multiple apps to tsuru to load test app deployment.

https://www.pivotaltracker.com/story/show/98091996

**What**

This PR sets up a load test for creating and deploying apps to a Tsuru platform endpoint. We make use of the SSH helper created as part of the integration tests, to enable us to deploy using `git push` as part of the deployment load test. We have used https://github.com/flood-io/ruby-jmeter as a DSL to speed up the writing of tests. However, in this usecase we have an issue with calling an os process sampler (a way of running a command outside of the load test). 

**How to test**

Clone the repo and then follow the README.md setup instructions (as we need to test those as part of the PR) then run a set of deploy apps and cleanup apps scripts. You should get a set of results from jmeter to prove that the tests have been run and output from running `tsuru app-list` will show you the created apps after running the `deploy_apps.rb` generated jmeter run. 

**Who can test**

Anyone on the core team apart from @saliceti
